code/modules/tgui_input/text.dm

```diff
@@ -32,9 +32,9 @@
 	if(!user.client.prefs.read_preference(/datum/preference/toggle/tgui_input))
 		if(encode)
 			if(multiline)
-				return stripped_multiline_input(user, message, title, default, max_length)
+				return stripped_multiline_input(user, message, title, default, max_length + 1)
 			else
-				return stripped_input(user, message, title, default, max_length)
+				return stripped_input(user, message, title, default, max_length + 1)
```

Comment:
i know the DM bluebook declares this, but this heavily relies on institutional knowledge/people to actually be aware of this. if you think it's a good idea, maybe we could make a new define + autodoc that with something like:

```dm
#define PREVENT_CHARACTER_TRIM_LOSS(integer) (integer + 1)
```

we can then autodoc this define and people wouldn't wonder why we have such a weird inconsistency/have to pull up the blame to figure out why you did such a decision, it's a much shorter closed loop. if you don't think it's worth the while then it's alright, but I figured I'd mention it since it came into my mind.

Reply:
Sure, where would you locate this define? def/undef inside the file or in an appropriate file in the defines folder?

Reply:
Defines folder because it can be used in a global context